src/io/export_stl.cc

```diff
@@ -39,123 +41,144 @@
 
 namespace {
 
-std::string toString(const Vector3d& v)
+/* Define values for double-conversion library. */
+#define DC_BUFFER_SIZE (128)
+#define DC_FLAGS (double_conversion::DoubleToStringConverter::UNIQUE_ZERO | double_conversion::DoubleToStringConverter::EMIT_POSITIVE_EXPONENT_SIGN)
```

Comment:
I know this is basically copied verbatim from Value.cc, but I would remove the `EMIT_POSITIVE_EXPONENT_SIGN` flag here.  I was planning to remove that one from the original code too.  Also meant to convert all these to `constexpr` instead of defines.

See this [snippet from my unmerged PR](https://github.com/openscad/openscad/pull/4384/files#diff-796468af57e236dd8c69179c2126ae82e0c22250452807322bc3dd53d2a9b336R14-R23) for the converted values.  I had also tweaked a couple of other values, but I don't know if those are relevant to `ToShortestSingle` (or `ToShortest`) without reviewing the `double-conversion` "docs" (header file comments).

I basically halted work on that PR to start implementing libfmt, but that ended up being much more work than anticipated.

Reply:
Ah, cool PR! I've dropped EMIT_POSITIVE_EXPONENT_SIGN and will keep the defines consistent-ish for now (unless you'd prefer I constexprify both places in this PR).

Re/ libfmt, once all compilers fully support C++20 (clang 14 doesn't seem there yet), would we get enough of its features through std::format?